zxfer

```diff
@@ -616,7 +616,7 @@ grep ^$source@) > /dev/null 2>&1
 
         # since the snap was matched, let's not waste our time searching for 
 	# it again
-        zfs_source_snaps=$(echo "$zfs_source_snaps" | grep -v "$src_snap")
+        zfs_source_snaps=$(echo "$zfs_source_snaps" | grep -v "$src_snap"$)
```

Comment:
might be worth enforcing that it doesn't leak on the other end too
```suggestion
        zfs_source_snaps=$(echo "$zfs_source_snaps" | grep -v @"$src_snap"$)
```

Reply:
My first instinct was `^` to anchor it, but then I thought the snapshot name would be `dataset@snapshot`, but `$zfs_source_snaps` will already have been processed via zfs list | cut etc, so yes, `^` is correct.